bundles/org.jkiss.utils/src/org/jkiss/utils/BeanUtils.java

```diff
@@ -366,4 +369,21 @@ public static Object invokeStaticMethod(Class<?> objectType, String name, Class<
         }
     }
 
+    /**
+     * Attempts to find a specified class from the
+     * supplied instance object of its inheritor.
+     *
+     * @param instance  an instance to locate the class from
+     * @param className a name of required class to locate
+     * @return the class object or <code>null</code> if <code>instance</code> does not inherit from specified class
+     */
+    @Nullable
+    public static Class<?> findClassFromInstance(@NotNull Object instance, @NotNull String className) {
+        for (Class<?> cls = instance.getClass(); cls != null; cls = cls.getSuperclass()) {
+            if (cls.getName().equals(className)) {
+                return cls;
+            }
+        }
+        return null;
+    }
```

Comment:
I am forced to say that neither the name of this util method nor its comments are informative. It's absolutely clear from the code that it finds an ancestor with the specified `className`, but unfortunately, there is nothing about that in the name or the comments.

Although the method is quite simple, it would be nice to see a small unit test.

I also believe that `{@code}` blocks are substantially more readable that `<code></code>`.

Reply:
It may not be an ancestor, e.g. supplied `instance` is already an instance of required class. Suggest better name.

Reply:
isInstanceOf would be better

+ move class name to constant
+ include class name in the exception message

Reply:
The purpose of this method is to locate concrete class named `className` from the supplied `instance`, so `isInstanceOf` does not reflect that and sounds like we want to produce a boolean result. We decided to use name `findAncestorClass` instead.

The other two notes are fair.